src/Donors/resources/admin/components/DonorDetailsPage/Tabs/Overview.tsx

```diff
@@ -187,10 +188,13 @@ export default function DonorDetailsPageOverviewTab() {
                     <Header
                         title={__('Private Note', 'give')}
                         subtitle={__('This note will be seen by only admins', 'give')}
-                        actionOnClick={() => {}}
+                        actionOnClick={() => setIsAddingNote(true)}
                         actionText={__('Add note', 'give')}
                     />
-                    <PrivateNote />
+                    <PrivateNotes
+                        donorId={donorId}
+                        context={[isAddingNote, setIsAddingNote]}
+                    />
```

Comment:
Just a thought here: we’re creating a piece of state there, but it’s only being used inside the `PrivateNotes` component. That kind of breaks encapsulation and makes `PrivateNotes` dependent on its parent to function properly.

Maybe it would make sense to move the `Header` component (and the state) into `PrivateNotes` so everything stays self-contained? That way, the state logic would live where it’s actually needed, and PrivateNotes could be reused more easily elsewhere.

Reply:
@pauloiankoski I agree with you, but the thing is, we are triggering adding note functionality outside the private notes component. That's why I need both state and state handler inside private notes component. I mean, I can change this to use two attributes instead of just context.

Reply:
My suggestion was to do a full-encapsulation into the PrivateNotes component so that it contains everything it needs to work (aka move Header and state into PrivateNotes)

Reply:
Resolved: https://github.com/impress-org/givewp/pull/7959/commits/1b8f39aeb820c30fdb7af9a5af2628f370dae389